public/app/features/alerting/unified/components/rule-viewer/v2/RuleViewer.v2.tsx

```diff
@@ -49,124 +90,360 @@ const RuleViewer = ({ match }: RuleViewerProps) => {
+  const rulesPermissions = getRulesPermissions(identifier.ruleSourceName);
+  const hasCreateRulePermission = contextSrv.hasPermission(rulesPermissions.create);
+  const canSilence = useCanSilence(rule);
+
   if (loading) {
     return <LoadingPlaceholder text={'Loading...'} />;
   }
 
+  // TODO improve error handling here
   if (error) {
-    return String(error);
+    if (typeof error === 'string') {
+      return error;
+    }
+
+    return <Alert title={'Uh-oh'}>Something went wrong loading the rule</Alert>;
   }
 
   if (rule) {
-    const summary = rule.annotations['summary'];
+    // should fit in a tweet
+    const summary = rule.annotations[Annotation.summary];
     const promRule = rule.promRule;
 
     const isAlertType = isAlertingRule(promRule);
-    const numberOfInstance = isAlertType ? promRule.alerts?.length : undefined;
+    const isGrafanaManagedRule = isGrafanaRulerRule(rule.rulerRule);
+    const numberOfInstance = isAlertType ? (promRule.alerts ?? []).length : undefined;
 
     const isFederatedRule = isFederatedRuleGroup(rule.group);
     const isProvisioned = isGrafanaRulerRule(rule.rulerRule) && Boolean(rule.rulerRule.grafana_alert.provenance);
 
+    /**
+     * Since Incident isn't available as an open-source product we shouldn't show it for Open-Source licenced editions of Grafana.
+     * We should show it in development mode
+     */
+    const shouldShowDeclareIncidentButton = !isOpenSourceEdition() || isLocalDevEnv();
+    const shareUrl = createShareLink(rule.namespace.rulesSource, rule);
+
+    const copyShareUrl = () => {
+      if (navigator.clipboard) {
+        navigator.clipboard.writeText(shareUrl);
+        appEvents.emit(AppEvents.alertSuccess, ['URL copied to clipboard']);
+      }
+    };
+
     return (
       <>
-        <Stack direction="column" gap={3}>
+        <Stack direction="column" gap={2}>
           {/* breadcrumb and actions */}
-          <Stack>
-            <BreadCrumb folder={rule.namespace.name} evaluationGroup={rule.group.name} />
-            <Spacer />
-            <Stack gap={1}>
-              <Button variant="secondary" icon="pen">
-                Edit
-              </Button>
-              <Button variant="secondary">
-                <Stack alignItems="center" gap={1}>
-                  More <Icon name="angle-down" />
+          <BreadCrumbs folder={rule.namespace.name} evaluationGroup={rule.group.name} />
+
+          <Stack direction="column" gap={2}>
+            {/* header */}
+            <Stack direction="column" gap={1}>
+              <Stack direction="row" alignItems="center">
+                {/* // TODO normalize states from prom to grafana and pass to title */}
+                <Title name={rule.name} state={isAlertType ? promRule.state : undefined} />
+                <Spacer />
+                <Stack gap={1}>
+                  {isEditable && <EditButton identifier={identifier} />}
+                  <Dropdown
```

Comment:
Would be nice to use the standard Page component here but with custom title rendering? And having two breadcrumbs on top of each other feels very wrong.

Reply:
The standard page component has actions, and supports custom title & subtitle rendering, 
thinking of making those actions sticky as you scroll down. So would be good to be able to reuse that here if it makes sense.

Reply:
I asked in #grafana-frontend-platform what they think (They own nav & Page templates), I don't really know what is best here, using breadcrumbs to back to parent page with different filters or showing that some different way. 

But feels like there should be a better alternative than showing what looks like breadcrumbs at the top under the main breadcrumbs